nodejs/eks/cluster.ts

```diff
@@ -322,10 +322,8 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     }
 
     if (args.nodeGroupOptions && (
-        args.nodeSubnetIds ||
         args.nodeAssociatePublicIpAddress ||
         args.instanceType ||
-        args.instanceProfileName ||
```

Comment:
Why are these vars being removed?

Reply:
This logic makes it the code throw an exception if any of the vars in the second group are specified when nodeghroupoptions are specified. We do specify nodegroupoptions along with subnetids and instanceprofilename.  I think I was unable to exlude these from the main options and only set them in the nodeGrioupOptions. In any case, this is not related to the eniconfig change tho, so I can remove it from this PR and revisit separately.

Reply:
Ive removed the unrelated changes.